datafusion/core/tests/parquet/row_group_pruning.rs

```diff
@@ -1264,3 +1264,37 @@ async fn prune_periods_in_column_names() {
         .test_row_group_prune()
         .await;
 }
+
+#[tokio::test]
+async fn test_row_group_all_null_values() {
+    // Three row groups:
+    // 1. all Null values
+    // 2. values from 1 to 5
+    // 3. all Null values
+
+    // After pruning, only row group 2 should be selected
+    RowGroupPruningTest::new()
+        .with_scenario(Scenario::AllNullValues)
+        .with_query("SELECT * FROM t WHERE \"i8\" <= 5")
+        .with_expected_errors(Some(0))
+        .with_matched_by_stats(Some(1))
+        .with_pruned_by_stats(Some(2))
+        .with_expected_rows(5)
+        .with_matched_by_bloom_filter(Some(0))
+        .with_pruned_by_bloom_filter(Some(0))
+        .test_row_group_prune()
+        .await;
+
+    // After pruning, only row group 1,3 should be selected
+    RowGroupPruningTest::new()
+        .with_scenario(Scenario::AllNullValues)
+        .with_query("SELECT * FROM t WHERE \"i8\" is Null")
```

Comment:
Could you also add a tests:
1. `i16 IS NOT NULL` (to cover the opposite)
2. `i32 > 7` (prune via nulls and some via counts)

Reply:
@alamb thanks!  Add test in https://github.com/apache/arrow-datafusion/pull/9989/commits/11567d9f68e0e2e0459cf784f6f51f8e0e568912, and support the isNotNull, as the page level prune i prefer in next pr to keep this pr short and clean.